server/src/http_login.go

```diff
@@ -131,6 +132,35 @@ func httpLogin(c *gin.Context) {
 				http.Error(w, "That is not the correct password.", http.StatusUnauthorized)
 				return
 			}
+
+			// If they have provided a new password, replace the old one
+			if data.NewPassword != "" {
+				var passwordHash string
+				if v, err := argon2id.CreateHash(data.NewPassword, argon2id.DefaultParams); err != nil {
+					logger.Error("Failed to create a hash from the submitted password for " +
+						"\"" + data.Username + "\": " + err.Error())
+					http.Error(
+						w,
+						http.StatusText(http.StatusInternalServerError),
+						http.StatusInternalServerError,
+					)
+					return
+				} else {
+					passwordHash = v
+				}
```

Comment:
Isn't it cleaner not to introduce a mutable variable?
```suggestion
				passwordHash, err := argon2id.CreateHash(data.NewPassword, argon2id.DefaultParams)
				if err != nil {
					logger.Error("Failed to create a hash from the submitted password for " +
						"\"" + data.Username + "\": " + err.Error())
					http.Error(
						w,
						http.StatusText(http.StatusInternalServerError),
						http.StatusInternalServerError,
					)
					return
				}
```

Reply:
no, because now you are scoping err inproperly

Reply:
the rest of the function should never have access to err